scripts/jenkins/ardana/ansible/init.yml

```diff
@@ -156,6 +156,23 @@
       src: input-model-firewall-rules.yml
       dest: /var/lib/ardana/openstack/my_cloud/definition/data/firewall_rules.yml
 
+  - name: Install the patch command
+    zypper:
+        name: patch
+        state: preset
+    # https://gist.github.com/marcusphi/6791404
+    when: patches is defined and (not patches|trim == '')
+
+  - name: Curl the patches and apply them
+    become: true
+    become_user: ardana
+    shell: |
+        curl -k '{{ item }}' | patch -p1
```

Comment:
I had a similar idea but I thought it would be nice to use gerrit changes: https://github.com/SUSE-Cloud/automation/pull/2405/files#diff-aa8d8108e8195416bc8f3a867489d4b5R23

Could we do something like that? Gerrit offers raw patches but they are either base64 encoded or zipped and you have to know the commit hash in addition to the change number.

Reply:
I think that the idea is only to provide the URL like this

```
curl -sk "https://gerrit.suse.provo.cloud/gitweb?p=ardana/osconfig-ansible.git;a=commitdiff_plain;h=393c355364add4c629d4123416866552b809031a" | less
```

Reply:
Whatever you decide on, please make it user-friendly as possible.  Being forced to navigate through gitweb to find the exact patch URL does not count as user-friendly ;-)

Reply:
> I think that the idea is only to provide the URL like this

TIL this exists

Reply:
@aspiers well now you know `gitweb >> raw` patch :)
@cmurphy you can use `current` word as `revsion-id` for the change hash something like `https://gerrit.suse.provo.cloud/changes/2985/revisions/current/patch?zip` and 
`item` can always be 
```
https://gerrit.suse.provo.cloud/changes/2985/revisions/current/patch?zip | gunzip
```

Reply:
@sjamgade Sorry, I'm not sure what you mean?  But please see my above point that having to manually construct patch URLs is not user-friendly, so it would be great to avoid that.

Reply:
by not auto creating gerrit change urls, I can host the patches anywhere I want like I showed in my comment below

Reply:
@aspiers but then it forces only the latest patcheset to be used. What do you think of having two different variables for each use case(one just having gerrit ids, and another for raw), user friendly enough ?

Reply:
> @aspiers but then it forces only the latest patcheset to be used.

No it doesn't, that was just an example.  If only the change id is specified it could take the latest patchset, but if something like `1234/2` is specified then it could take the 2nd patchset.

>What do you think of having two different variables for each use case(one just having gerrit ids, and another for raw), user friendly enough ?

That would also work but personally I think a single multi-purpose parameter is a bit cleaner.  So to clarify, it would accept all of the following:

- `1234`
- `1234/3`
- `https://path/to/my/patch`